pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -527,7 +510,7 @@ where
 {
     backoff::retry(
```

Comment:
Would like to understand a reason for doing that; I just intentionally added it.

At least theoretically we have multiple error types at play with `backoff::retry`, and alternatives are to couge/claw the cancellation from within anyhow or random error type (next PR has usage of std::io::Error for this). Whenever I see an `Option<T>` it could also mean `Result<T, ()>`, latter just is shunned by clippy. Take for example `std::num::NonZeroUsize::new` return type is not `Result<NonZeroUsize, usize>` but just `Option<NonZeroUsize>`.

Reply:
We chatted about this. No big revelations there, we quickly moved on :) Resolving.